wdl/ExpansionHunterDenovo.wdl

```diff
@@ -0,0 +1,347 @@
+      reference_filename = reference_filename,
+      ehdn_docker = ehdn_docker,
+      runtime_attr = runtime_attr_merge,
+      postfixes = postfixes
+  }
+
+  call STRAnalyze {
+    input:
+      analysis_type = analysis_type,
+      str_comparison_type = str_comparison_type,
+      manifest = Merge.manifest,
+      multisample_profile = Merge.multisample_profile,
+      ehdn_docker = ehdn_docker,
+      runtime_attr = runtime_attr_analysis
+  }
+
+  output {
+    Array[File] cases_locus = CasesProfiles.locus
+    Array[File] cases_motif = CasesProfiles.motif
+    Array[File] cases_str_profile = CasesProfiles.str_profile
+    Array[File] controls_locus = ControlsProfiles.locus
+    Array[File] controls_motif = ControlsProfiles.motif
+    Array[File] controls_str_profile = ControlsProfiles.str_profile
+    File multisample_profile = Merge.multisample_profile
+    Array[File] analysis_results = STRAnalyze.results
+  }
+}
+
+task ComputeSTRProfile {
+  input {
+    String filename
+    File reads_filename
+    File index_filename
+    File reference_filename
+    Int min_anchor_mapq
+    Int max_irr_mapq
+    String ehdn_docker
+    RuntimeAttr runtime_attr
+    FilenamePostfixes postfixes
+  }
+
+  output {
+    File locus = "${filename}${postfixes.locus}"
+    File motif = "${filename}${postfixes.motif}"
+    File str_profile = "${filename}${postfixes.profile}"
+  }
+
+  command <<<
+
+    ExpansionHunterDenovo profile \
+    --reads ~{reads_filename} \
+    --reference ~{reference_filename} \
+    --output-prefix ~{filename} \
+    --min-anchor-mapq ~{min_anchor_mapq} \
+    --max-irr-mapq ~{max_irr_mapq}
+
+  >>>
+
+  runtime {
+    docker: ehdn_docker
+    cpu: runtime_attr.cpu_cores
+    memory: runtime_attr.mem_gb + " GiB"
+    disks: "local-disk " + runtime_attr.disk_gb + " HDD"
+    bootDiskSizeGb: runtime_attr.boot_disk_gb
+    preemptible: runtime_attr.preemptible_tries
+    maxRetries: runtime_attr.max_retries
+  }
+}
+
+task Merge {
+  input {
+    Array[File] cases
+    Array[File] controls
+    File reference_filename
+    String ehdn_docker
+    RuntimeAttr runtime_attr
+    FilenamePostfixes postfixes
+  }
+
+  output {
+    File manifest = "manifest.tsv"
+    File multisample_profile = "${output_prefix}${postfixes.merged_profile}"
+  }
+
+  Int cases_length = length(cases)
+  Int controls_length = length(controls)
+  String output_prefix = "merged"
+
+  # This shell script uses EHdn's `merge` command
+  # to merge STR profiles on all the individual samples
+  # into a single JSON file.
+  #
+  # The script is composed of two parts:
+  # - Create a "manifest" file based on the given inputs.
+  #   The manifest file is composed of three columns:
+  #   (1) Sample name (infered from the filename);
+  #   (2) case/control (infered from files in the inputs
+  #       `cases` and `controls`);
+  #   (3) file path.
+  #
+  # - Call EHdn's `merge` method using the input and
+  #   the generaged manifest file.
+  command <<<
```

Comment:
Always start this section with `set -euo pipefail` (or `set -euxo pipefail` if it won't generate excessive logging).

Reply:
Good point, I added `set -euxo pipefail` to all the commands.